web/packages/teleterm/src/services/pty/ptyHost/buildPtyOptions.ts

```diff
@@ -203,3 +255,23 @@ function getKubeConfigFilePath(
 ): string {
   return path.join(settings.kubeConfigsDir, command.kubeConfigRelativePath);
 }
+
+async function resolveShell(
+  cmd: ShellCommand,
+  settings: RuntimeSettings,
+  ptyOptions: PtyOptions
+): Promise<Shell | undefined> {
+  if (cmd.shellId !== CUSTOM_SHELL_ID) {
+    return settings.availableShells.find(s => s.id === cmd.shellId);
+  }
+
+  const { customShellPath } = ptyOptions;
+  if (customShellPath) {
+    try {
+      await fs.access(customShellPath);
```

Comment:
Why do we need to check access first?

Reply:
I wanted to gracefully handle a situation when someone sets `customShellPath` to non-existing path. I then run the default shell and show a notification about the problem.
But now that I think about it, it is probably unnecessary and doesn't help much the user. We already show a message about non-existing executable when we try to actually run it. Let me remove this check.